File: bot/src/main/java/com/yauhenl/pibot/control/action/impl/Actions.java
package com.yauhenl.pibot.control.action.impl;

import com.yauhenl.pibot.control.action.Action;
import com.yauhenl.pibot.hardware.cascade.motor.Motor;
import org.springframework.stereotype.Service;

@Service
public class Actions {
    private static final int BACKWARD_SHIFT_VAL = 156;
    private static final int FORWARD_SHIFT_VAL = 99;
    private static final int LEFT_SHIFT_VAL = 46;
    private static final int RIGHT_SHIFT_VAL = 209;
    private static final int STOP_SHIFT_VAL = 0;

    public static Action moveBackwardAction = botControl -> {
        botControl.getCascade().getShift().writeShift(BACKWARD_SHIFT_VAL);
    };

    public static Action moveForwardAction = botControl -> {
        botControl.getCascade().getShift().writeShift(FORWARD_SHIFT_VAL);
    };

    public static Action moveLeftAction = botControl -> {
        botControl.getCascade().getShift().writeShift(LEFT_SHIFT_VAL);
    };

    public static Action moveRightAction = botControl -> {
        botControl.getCascade().getShift().writeShift(RIGHT_SHIFT_VAL);
    };

    public static Action stopMovingAction = botControl -> {
        botControl.getCascade().getShift().writeShift(STOP_SHIFT_VAL);
    };

    public static Action allMotorsPowerOffAction = botControl -> botControl.getCascade().getMotors().values().forEach(Motor::setLowOutput);

    public static Action allMotorsPowerOnAction = botControl -> botControl.getCascade().getMotors().values().forEach(Motor::setHighOutput);
}

File: bot/src/main/kotlin/com/yauhenl/pibot/control/Actions.kt
package com.yauhenl.pibot.control

import com.yauhenl.pibot.hardware.Motor
import com.yauhenl.pibot.service.BotControl

object Actions {
    private val BACKWARD_SHIFT_VAL = 156
    private val FORWARD_SHIFT_VAL = 99
    private val LEFT_SHIFT_VAL = 46
    private val RIGHT_SHIFT_VAL = 209
    private val STOP_SHIFT_VAL = 0

    val moveBackwardAction = { botControl: BotControl -> botControl.shiftRegister.writeShift(BACKWARD_SHIFT_VAL) }
    val moveForwardAction = { botControl: BotControl -> botControl.shiftRegister.writeShift(FORWARD_SHIFT_VAL) }
    val moveLeftAction = { botControl: BotControl -> botControl.shiftRegister.writeShift(LEFT_SHIFT_VAL) }
    val moveRightAction = { botControl: BotControl -> botControl.shiftRegister.writeShift(RIGHT_SHIFT_VAL) }
    val stopMovingAction = { botControl: BotControl -> botControl.shiftRegister.writeShift(STOP_SHIFT_VAL) }
    val allMotorsPowerOffAction = { botControl: BotControl -> botControl.motors.values.forEach(Motor::setLowOutput) }
    val allMotorsPowerOnAction = { botControl: BotControl -> botControl.motors.values.forEach(Motor::setHighOutput) }
}